test/safe.test.ts

```diff
@@ -101,3 +114,81 @@ it("Unable to authorize overlapping batches", async () => {
   await swap(0, 10);
   await expect(swap(1, 9)).to.be.rejectedWith("Batches must be gapless");
 });
+
+it("Able to prove fraud", async () => {
```

Comment:
nit but I think we could tweak the wording slightly.
```suggestion
it("can handle a fraud proof", async () => {
```